client/src/reducers/autosave.ts

```diff
@@ -0,0 +1,92 @@
+import type { AnyAction } from "redux";
+
+import type AnnoMatrix from "../annoMatrix/annoMatrix";
+import type { RootState } from ".";
+import type { Genesets } from "./genesets";
+
+export interface AutosaveState {
```

Comment:
Reducer now tracks last-saved annoMatrix and genesets alongside in-flight flags; useful map for anyone checking how autosave decisions are made.